### PR TITLE
Webserver port

### PR DESCRIPTION
**Pull request by Skylark13**
This PR also contains the changes from my previous PR. For some reason, even though I created separate branches for them, this branch ended up with the cumulative changes...

You should only merge commit 3a4e1b0, that contains the changes related to making the webserver port settable.

### CODE REVIEW

File: microprofile.cpp [hunk 1]
```diff
@@ -5003,16 +5015,16 @@ void MicroProfileWebServerStart()
 
 
 
-	S.nWebServerPort = (uint32_t)-1;
+	int startPort = S.nWebServerPort;
```
Comment (jonasmr) [inline microprofile.cpp]: sorry for being nitpicky, but use nStartPort please

File: microprofile.h [hunk 2]
```diff
@@ -642,6 +642,7 @@ MICROPROFILE_API void MicroProfileSetCurrentNodeVulkan(uint32_t nNode);
 MICROPROFILE_API void MicroProfileDumpFile(const char* pHtml, const char* pCsv, float fCpuSpike, float fGpuSpike);
 MICROPROFILE_API void MicroProfileDumpFileImmediately(const char* pHtml, const char* pCsv, void* pGpuContext);
 MICROPROFILE_API uint32_t MicroProfileWebServerPort();
+MICROPROFILE_API void MicroProfileSetWebServerPort(int port);
```
Comment (jonasmr) [inline microprofile.h]: another nit: nPort instead of port

File: microprofile.cpp [hunk 3]
```diff
@@ -3819,6 +3820,17 @@ uint32_t MicroProfileWebServerPort()
 	return S.nWebServerPort;
 }
 
+void MicroProfileSetWebServerPort(int port)
```
Comment (jonasmr) [inline microprofile.cpp]: nit: nPort

**Comment by Skylark13** [comment]
Should be good now. Sorry again for the fact this PR contains the previous PR's changes too.

**Comment by jonasmr** [comment]
looks good, thank you.

**Pull request merged**